libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -217,21 +235,56 @@ void AC_WPNav::loiter_soften_for_landing()
 
     // set target position to current position
     _pos_control.set_xy_target(curr_pos.x, curr_pos.y);
-    _pos_control.freeze_ff_xy();
 }
 
 /// set_pilot_desired_acceleration - sets pilot desired acceleration from roll and pitch stick input
-void AC_WPNav::set_pilot_desired_acceleration(float control_roll, float control_pitch)
+void AC_WPNav::set_pilot_desired_acceleration(float euler_roll_angle_cd, float euler_pitch_angle_cd)
 {
-    // convert pilot input to desired acceleration in cm/s/s
-    _pilot_accel_fwd_cms = -control_pitch * _loiter_accel_cmss / 4500.0f;
-    _pilot_accel_rgt_cms = control_roll * _loiter_accel_cmss / 4500.0f;
+    float nav_dt = 0.0025f;
```

Comment:
txs.  const-ified all local vars in get_pilot_desired_acceleration method.

Reply:
Can we set this based on the update rate rather than hard coded.

Reply:
OK, new commit added that adds a dt parameter to AC_WPNav::set-pilot-desired-acceleration().  I found a couple of issues with RTL and Land not passing in the desired lean-angles but instead the rc input directly so that's fixed as well.